### PR TITLE
Fix bug with continuous legends

### PR DESCRIPTION
**Pull request by dan-knight**
## Description

In new versions of R (> R v4.3.0), lattice does not allow `labels` to be specified in a `draw.colorkey` without corresponding values for `at`. This change fixes BPG's call to `draw.colorkey` within  `legend.grob`, placing labels equidistant along the range of continuous values.

## Checklist

-   [X] This PR does NOT contain [PHI](https://ohrpp.research.ucla.edu/hipaa/) or germline genetic data. A repo may need to be deleted if such data is uploaded. Disclosing PHI is a [major problem](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m).

-   [X] This PR does NOT contain molecular files, compressed files, output files such as images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other non-plain-text files. To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example.

-   [X] I have read the [code review guidelines](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List).

-   [X] I have set up or verified the `main` branch protection rule following the [github standards](https://confluence.mednet.ucla.edu/pages/viewpage.action?spaceKey=BOUTROSLAB&title=GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

-   [X] The name of the branch is meaningful and well formatted following the [standards](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].

-   [X] I have added the major changes included in this pull request to the `NEWS` under the next release version or unreleased, and updated the date. I have also updated the version number in `DESCRIPTION` according to [semantic versioning](https://semver.org/) rules.

-   [X] Both `R CMD build` and `R CMD check` run successfully.

### CODE REVIEW

File: R/legend.grob.R [hunk 1]
```diff
@@ -142,6 +142,18 @@ legend.grob <- function(
 				legendi[['width']]  <- if (is.null(legendi[['width']])) { 2 } else { legendi[['width']] };
 
 				colorRamp <- colorRampPalette(legendi[['colours']]);
+				
+				labels.at <- if (is.null(legendi[['labels']])) legendi[['at']] else {
+				    n.labels <- length(legendi[['labels']]);
+				    max.value <- if (!is.null(legendi[['continuous.amount']])) legendi[['continuous.amount']] else 100;
+			        boundaries <- seq(0, max.value, length.out = n.labels + 1);
+
+			        sapply(
```
Comment (stefaneng) [inline R/legend.grob.R:151]: Indent off here at least on github maybe just spaces/tabs mixed?

**Review by stefaneng — COMMENTED** [review]
Do you have an example of this to test?

**Comment by dan-knight** [comment]
> Do you have an example of this to test?

I actually found out about this issue because of a failing example from `create.heatmap` (the last one, continuous legend). This example is run with our `R CMD check` pipeline, so the passing check shows that it now runs without errors. Here's the plot output with the fix:

image

File: R/legend.grob.R (hunk 1 above)
Comment (dan-knight) [reply R/legend.grob.R]: Seems like it! Should be fixed now.

Comment (stefaneng) [reply R/legend.grob.R]: Whitespace is still not good. There are a bunch of tabs before sapply

image

Comment (dan-knight) [reply R/legend.grob.R]: Screen Shot 2023-05-23 at 9 47 25 AM
Seems to be fixed now. (Desregard the refactor - indentation should be the same before vs. after.)

**Review by stefaneng — APPROVED** [review]

**Pull request merged**